explore/src/main/scala/explore/users/UserPreferencesPopup.scala

```diff
@@ -166,12 +139,27 @@ object UserPreferencesContent:
             "Delete",
             cell = { cell =>
               val keyId = cell.value.id
+
+              val action = for {
+                _ <-
+                  DeleteApiKey[IO].execute(keyId, modParams = props.vault.addAuthorizationHeaderTo)
+                _ <- newKey.set(NewKey(none)).toAsync
+              } yield ()
+
+              val deleteKey = deleteConfirmation(
+                s"This action will delete the key with id: $keyId. This action cannot be reversed.",
+                "Key delete",
+                "Yes, delete",
+                action,
+                isActive
+              )
+
               <.div(
                 ExploreStyles.ApiKeyDelete,
                 Button(
                   icon = Icons.Trash,
                   severity = Button.Severity.Secondary,
-                  onClick = deleteKey(keyId, isAdding, newKey, props.vault),
+                  onClick = deleteKey,
```

Comment:
Should this button be disabled when `isActive` is true, too?

Reply:
Not sure as is Active is used on  the popup for invites, though now that you mention it maybe it should be used while deleting too

Reply:
Sorry I got confused, you're right we should use it in the button
This is also because thee confirmation dialog method takes a `View[IsActive]`